src/WhereConditions/WhereConditionsBaseDirective.php

```diff
@@ -168,7 +168,7 @@ protected static function assertValidColumnReference(string $column): void
         // - must not start with a digit, dot or hyphen
         // - must contain only alphanumerics, digits, underscores, dots or hyphens
         // Dots are allowed to reference a column in a table: my_table.my_column.
-        $match = \Safe\preg_match('/^(?![0-9.-])[A-Za-z0-9_.-]*$/', $column);
+        $match = \Safe\preg_match('/^(?![0-9.-])([A-Za-z0-9_.-]|->)*$/', $column);
```

Comment:
When would this be necessary? Please provide a test case that fails without this change and passes with it.

Reply:
Is there a specific test model that I can modify adding a column or any is good?

Reply:
Any model with a string column should suffice, no?

Reply:
Nope, would need a model with a JSON column in the database

Reply:
Ok, just pick any one and add a nullable column.